core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
In fact, even the constants used by my patch are dangerous: now they force NettyTransport to be loaded, look at its static block, which then loads NettyUtils and then netty classes... (god knows what they are doing, now and in the future)

Previously this stuff was all initialized after security manager was loaded, so we didn't have to worry about sheisty things happening.

I'm just mentioning this to explain my reasoning.